S32K/libuavcan/media/S32K/canfd.hpp

```diff
@@ -0,0 +1,195 @@
+
+    /**
+     * Block with timeout for available Message buffers.
+     * @param [in]     timeout                  The amount of time to wait for and available message buffer.
+     * @param [in]     ignore_write_available   If set to true, will check availability only for RX MB's
+     *
+     * @return  libuavcan::Result::SuccessTimeout if timeout occurred and no required MB's became available.
+     *          libuavcan::Result::Success if an interface is ready for read, and if
+     *          @p ignore_write_available is false, or write.
+     */
+    virtual Result select(libuavcan::duration::Monotonic timeout, bool ignore_write_available) override;
+};
+
+/**
+ * Implementation of the methods from libuavcan's media layer abstracct class InterfaceManager,
+ * with the template arguments listed below; for further details of this interface class,
+ * refer to the template declaration in libuavcan/media/interface.hpp
+ *
+ * InterfaceGroupT    = S32K_InterfaceGroup  (previously declared class in the file)
+ * InterfaceGroupPtrT = S32K_InterfaceGroup* (raw pointer)
+ */
+class InterfaceManager : public media::InterfaceManager<InterfaceGroup, InterfaceGroup*>
+{
+private:
+    /** S32K_InterfaceGroup type object member, which address is used in the factory method next */
+    InterfaceGroupType InterfaceGroupObj_;
+
+public:
+    /**
+     * Initialize the peripherals needed for the driver in the target MCU, also configures the
+     * core clock sources to the Normal RUN profile,
+     * @param  filter_config         The filtering to apply equally to all FlexCAN instances.
+     * @param  filter_config_length  The length of the @p filter_config argument.
+     * @param  out_group             A pointer to set to the started group. This will be nullptr if the start method
+     *                               fails.
+     * @return libuavcan::Result::Success if the group was successfully started and a valid pointer was returned.
+     * @return libuavcan::Result::Failure if the initialization fails at some point.
+     *         The caller should assume that @p out_group is an invalid pointer if any failure is returned.
+     * @return libuavcan::Result::BadArgument if filter_config_length is out of bound.
+     */
+    virtual Result startInterfaceGroup(const typename InterfaceGroupType::FrameType::Filter* filter_config,
+                                       std::size_t                                           filter_config_length,
+                                       InterfaceGroupPtrType&                                out_group) override;
+
+    /**
+     * Release and deinitialize the peripherals needed for the current driver, disables all the FlexCAN
+     * instances available, waiting for any pending transmission or reception to finish before, also
+     * resets the LPIT timer used for timestamping, does not deconfigures the core clock sources
```

Comment:
```suggestion
     * resets the LPIT timer used for time-stamping, does not deconfigure the core clock sources
```

Reply:
Fixed in https://github.com/noxuz/platform_specific_components/commit/f86edb519ec3bd86aff1aade093e4e3084f4c009